src/npctalk.cpp

```diff
@@ -3218,84 +3221,82 @@ std::string give_item_to( npc &p, bool allow_use, bool allow_carry )
         return _( "Are you <swear> insane!?" );
     }
 
-    std::string no_consume_reason;
+    bool taken = false;
+    std::string reason = _( "Nope." );
+    int our_ammo = p.ammo_count_for( p.weapon );
+    int new_ammo = p.ammo_count_for( given );
+    const double new_weapon_value = p.weapon_value( given, new_ammo );
+    const double cur_weapon_value = p.weapon_value( p.weapon, our_ammo );
+    add_msg( m_debug, "NPC evaluates own %s (%d ammo): %0.1f",
+             p.weapon.type->get_id(), our_ammo, cur_weapon_value );
+    add_msg( m_debug, "NPC evaluates your %s (%d ammo): %0.1f",
+             given.type->get_id(), new_ammo, new_weapon_value );
```

Comment:
```suggestion
             given.typeId(), new_ammo, new_weapon_value );
```
Same for the user usage of `get_id()` above.